packages/core/src/clicks.ts

```diff
@@ -1,11 +1,25 @@
 import { defaultImpressionObserver } from './impressions'
+import { isFallback } from './fallback'
 import { DefaultLogger } from './logger'
 import { PrivateMozAdsImpressionObserver } from './private/types'
 import { MozAdsPlacementWithContent } from './types'
 
 const logger = new DefaultLogger({ name: 'core.clicks' })
 
 export async function recordClick(placement: MozAdsPlacementWithContent) {
+  // No need to send click events for fallback ads -- they are meant for an offline experience so
+  // they don't have dynamic click urls from MARS, so we can't register a click for them.
+  if (isFallback(placement)) return
+
+  const clickUrl = placement.content?.callbacks?.click
+  if (!clickUrl || !URL.canParse(clickUrl)) {
+    logger.error(`Invalid click URL for placement: ${placement.placementId}`, {
+      type: 'recordClick.invalidCallbackError',
+      eventLabel: 'invalid_url_error',
+      placementId: placement.placementId,
+    })
+    return
+  }
```

Comment:
This un-nests the `if-else`. Before, we had the `invalid_url_error` handled in the else, now we check for fallback, then url validity, up front and error and exit before getting to the happy path.